pkcs7/sign.go

```diff
@@ -11,40 +11,91 @@ import (
 	"crypto/sha256"
 	"crypto/x509"
 	"encoding/asn1"
+	"errors"
 	"io"
 	"time"
 )
 
-func Sign(r io.Reader, cert *x509.Certificate, priv *rsa.PrivateKey) ([]byte, error) {
-	hash := sha256.New()
-	if _, err := io.Copy(hash, r); err != nil {
+// Create Signature of message
+// message must be of type io.Reader or []byte
+// Returns signature and any error encountered.
+func Sign(message interface{}, certificate *x509.Certificate, privateKey *rsa.PrivateKey) ([]byte, error) {
```

Comment:
I agree with you that `interface{}` is not the best solution. 

Instead of using `io.Reader` I suggest to introduce an interface `Hashable` https://github.com/YannicBonenberger/gocrypto/commit/77cdc19b8129d4f803a98e7e7503d4d90cd51c16